### DISCUSSION TITLE
Docker registry: You must push your image using VERSION and not using SHA - why?

### DISCUSSION THREAD

**Discussion by 1player** [community/community#25202] [Actions]
<p>The title is taken verbatim from the Docker registry documentation: “You must push your image using <em>IMAGE_NAME:VERSION</em> and not using <em>IMAGE_NAME:SHA</em>” — why is that? We were actually planning on using the SHA for private-only packages.</p>
<p>Is that a requirement of the Github Packages system, or is that just a best practice and there is no problem for us to tag the images with the git sha?</p>

**Comment by Yanjingzhu** [comment]
<p>I tried in my side, use <strong>github.sha</strong> as docker image version could work. You can tag the images with the git sha.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/5/567c192dbc4398966bfe377c9cec44f59428996a.png" data-download-href="/uploads/short-url/cl4WoSvHeujbXzGftQikaEXhPgm.png?dl=1" title="GitHub sha.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181089374-53fcfbbc-beff-4958-bc4f-06162e01bc38.png" alt="GitHub sha.png" data-base62-sha1="cl4WoSvHeujbXzGftQikaEXhPgm" width="690" height="442" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/5/567c192dbc4398966bfe377c9cec44f59428996a_2_690x442.png, https://aws1.discourse-cdn.com/github/optimized/2X/5/567c192dbc4398966bfe377c9cec44f59428996a_2_1035x663.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/5/567c192dbc4398966bfe377c9cec44f59428996a.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/567c192dbc4398966bfe377c9cec44f59428996a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">GitHub sha.png</span><span class="informations">1061×680 55.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by 1player** [comment]
<p>Worked for me as well, I was just curious about the strong warning in the documentation. There must be a reason behind it I suppose.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>The SHA in document is just pointed to docker generated sha. </p>
<p>We currently do not support digests in the current version of Docker in Registry.</p>
<p>You can however use any sha you want as a tag other than the docker generated sha for that tag. Using the git sha should work perfectly fine as a tag is just a string.</p>
